swagger.yaml

```diff
@@ -3672,6 +6638,204 @@ definitions:
         type: string
         description: Array of image statuses.
 
+  agent-create-params:
+    type: object
+    required:
+      - agent_id
+    properties:
+      agent_id:
+        type: string
+        format: uuid
+      discovery_agent_version:
+        type: string
+
+  agent_registration_response:
+    allOf:
+        - $ref: '#/definitions/agent'
+        - properties:
+            next_step_runner_command:
+              type: object
+              description: Command for starting the next step runner
+              properties:
+                command:
+                  type: string
+                args:
+                  type: array
+                  items:
+                    type: string
+                retry_seconds:
+                  type: integer
+                  description: How long in seconds to wait before retrying registration if the command fails
+
+
+  agent:
+    type: object
+    required:
+      - kind
+      - id
+      - href
+      - status
+      - status_info
+    properties:
+      kind:
+        type: string
+        enum: ['Agent', 'AddToExistingClusterAgent']
+        description: |
+          Indicates the type of this object. Will be 'Agent' if this is a complete object or 'AgentLink' if it is just a link, or
+          'AddToExistingClusterAgent' for agent being added to existing OCP cluster, or
+      id:
+        type: string
+        format: uuid
+        description: Unique identifier of the object.
+        x-go-custom-tag: gorm:"primary_key"
+      href:
+        type: string
+        description: Self link.
+      infra_env_id:
+        type: string
+        format: uuid
+        x-go-custom-tag: gorm:"primary_key;foreignkey:InfraEnv"
+        description: The InfraEnv that this agent is associated with.
+      cluster_id:
```

Comment:
i think we should add a field of infra_env:
infra_env:
        type: object
        x-go-custom-tag: gorm:"foreignkey:ID;association_foreignkey:InfraEnvID"
        description: The InfraEnv that this agent is associated with.
        $ref: '#/definitions/infra_env'
So we get preloaded infra env from each host, will help us during ignition generation of the cluster

Reply:
We want to return the whole thing to the user?  Including nmstates?  And then in ListAgents we'll return it many times?  Doesn't sound right.

Reply:
The gorm will give you this behavior just because it was defined with foreign key

Reply:
we can move it to db, and remove it from swagger